dbt/include/fabric/macros/adapters/columns.sql

```diff
@@ -1,3 +1,15 @@
+{% macro fabric__get_empty_subquery_sql(select_sql, select_sql_header=none) %}
+    {% if sql.strip().lower().startswith('with') %}
+        {{ select_sql }}
+    {% else -%}
+        select * from (
+        {{ select_sql }}
+    ) dbt_sbq_tmp
+    where 1 = 0
+    {%- endif -%}
+
+{% endmacro %}
```

Comment:
this is very clever! can you add a docstring or comment for why this is needed?

Reply:
Some of the models are expressed as a CTE. SELECT * FROM (<CTE>) is not supported by Microsoft sql engines. So, I added a check to see if the sub query is CTE or not.